### PR TITLE
Upgrade to cosmwasm v0.16.0

### PR DESCRIPTION
**Pull request by MSNTCS**

### CODE REVIEW

File: contracts/anchor_beth_reward/src/user.rs [hunk 1]
```diff
@@ -49,60 +50,59 @@ pub fn handle_claim_rewards<S: Storage, A: Api, Q: Querier>(
...
+    }
+    .into();
+
+    Ok(Response::new()
+        .add_attributes(vec![
+            attr("action", "claim_reward"),
+            attr("holder_address", holder_addr),
+            attr("rewards", rewards),
+        ])
+        .add_submessage(SubMsg::new(bank_msg)))
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_reward/src/user.rs:84]: Maybe better to use add_message

File: contracts/anchor_beth_token/src/handler.rs [hunk 2]
```diff
@@ -1,257 +1,264 @@
...
-            send: vec![],
-        })],
-        log: res.log,
-        data: None,
-    })
+            funds: vec![],
+        })),
+    ];
+    Ok(Response::new()
+        .add_submessages(messages)
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:53]: If you don't need reply, add-message is better

File: contracts/anchor_beth_token/src/handler.rs [hunk 3]
```diff
@@ -1,257 +1,264 @@
...
+        contract_addr: reward_contract.to_string(),
+        msg: to_binary(&DecreaseBalance {
+            address: sender.to_string(),
+            amount,
+        })
+        .unwrap(),
+        funds: vec![],
+    }))];
+    Ok(Response::new()
+        .add_submessages(messages)
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:79]: Same

File: contracts/anchor_beth_token/src/handler.rs [hunk 4]
```diff
@@ -1,257 +1,264 @@
...
-            ],
-            res.messages,
-        ]
-        .concat(),
-        log: res.log,
-        data: None,
-    })
+    let res: Response = cw20_mint(deps, env, info, recipient.clone(), amount)?;
+    Ok(Response::new()
+        .add_submessages(vec![SubMsg::new(CosmosMsg::Wasm(WasmMsg::Execute {
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:96]: Same

File: contracts/anchor_beth_token/src/handler.rs [hunk 5]
```diff
@@ -1,257 +1,264 @@
...
         ],
-        log: res.log,
-        data: None,
-    })
+        res.messages,
+    ]
+    .concat();
+
+    Ok(Response::new()
+        .add_submessages(messages)
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:148]: Same

File: contracts/anchor_beth_token/src/handler.rs [hunk 6]
```diff
@@ -1,257 +1,264 @@
...
+            msg: to_binary(&IncreaseBalance {
+                address: recipient,
+                amount,
+            })
+            .unwrap(),
+            funds: vec![],
+        })),
+    ];
+    Ok(Response::new()
+        .add_submessages(messages)
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:188]: Same

File: contracts/anchor_beth_token/src/handler.rs [hunk 7]
```diff
@@ -1,257 +1,264 @@
...
+        contract_addr: reward_contract.to_string(),
+        msg: to_binary(&DecreaseBalance {
+            address: valid_owner.to_string(),
+            amount,
+        })
+        .unwrap(),
+        funds: vec![],
+    }))];
+    Ok(Response::new()
+        .add_submessages(messages)
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:216]: Same

File: contracts/anchor_beth_token/src/handler.rs [hunk 8]
```diff
@@ -1,257 +1,264 @@
...
-                }),
-            ],
-            res.messages,
-        ]
-        .concat(),
-        log: res.log,
-        data: None,
-    })
+    Ok(Response::new()
+        .add_submessages(messages)
```
Comment (YunSuk-Yeo) [inline contracts/anchor_beth_token/src/handler.rs:262]: Same

File: packages/beth/Cargo.toml [hunk 9]
```diff
@@ -17,20 +17,17 @@ incremental = false
...
+backtraces = ["cosmwasm-std/backtraces"]
 
 [dependencies]
-cw20 = { version = "0.2" }
-cosmwasm-storage = { version = "0.10.1" }
-terra-cosmwasm = { version = "1.2.2" }
-cosmwasm-std = { version = "0.10.1" }
-schemars = "0.7"
+cw20 = { version = "0.8.0" }
+cosmwasm-storage = { version = "0.16.0", features = ["iterator"] }
```
Comment (YunSuk-Yeo) [inline packages/beth/Cargo.toml:26]: Iterator is default now, so you can skip feature def

**Review by YunSuk-Yeo — APPROVED** [review]
Overall work lgtm

**Review by YunSuk-Yeo — APPROVED** [review]

**Review by YunSuk-Yeo — APPROVED** [review]

**Review by csanti — APPROVED** [review]

**Pull request merged**